server/src/main/java/org/opensearch/gateway/remote/ClusterStateDiffManifest.java

```diff
@@ -563,7 +564,16 @@ public static class Builder {
         private List<String> clusterStateCustomUpdated;
         private List<String> clusterStateCustomDeleted;
 
-        public Builder() {}
+        public Builder() {
+            customMetadataUpdated = Collections.emptyList();
```

Comment:
Just noticed that in the constructors, we are simply building the object with the normal `ArrayList`. We should be wrapping all the list objects with `Collections.unmodifiableList`

Reply:
Updated in latest changes